apps/examples/webserver_test/webserver_test_main.c

```diff
@@ -0,0 +1,1308 @@
+	printf("   $ webserver_test 3 GET https://127.0.0.1/ test_entity=5\n");
+	FUNC_EX;
+}
+
+static char* ws_test_wget_strcpy(char *dest, const char *src, struct http_client_request_t *ws)
+{
+	FUNC_EN;
+	int len = strlen(src);
+
+	if (dest + len - ws->buffer >= ws->buflen) {
+		PRNT("Error: buffer is too small");
+		return NULL;
+	}
+
+	memcpy(dest, src, len);
+	dest[len] = '\0';
+	FUNC_EX;
+	return dest + len;
+}
+
+static char* ws_test_wget_strlencpy(char *dest, const char *src, int len, struct http_client_request_t *ws)
+{
+	FUNC_EN;
+	if (dest + len - ws->buffer >= ws->buflen) {
+		PRNT("Error: buffer is too small");
+		return NULL;
+	}
+
+	memcpy(dest, src, len);
+	dest[len] = '\0';
+	FUNC_EX;
+	return dest + len;
+}
+
+static char* ws_test_wget_chunksize(char *dest, int len, struct http_client_request_t *ws)
+{
+	FUNC_EN;
+	char size[10] = {0,};
+
+	if (!dest) {
+		return NULL;
+	}
+
+	snprintf(size, sizeof(size), "%x", len);
+	len = strlen(size);
+	if (dest + len - ws->buffer >= ws->buflen) {
+		PRNT("Error: buffer is too small");
+		return NULL;
+	}
+	memcpy(dest, size, len);
+	FUNC_EX;
+	return dest + len;
+}
+
+int ws_test_http_client_response_init(struct http_client_response_t *response)
+{
+	FUNC_EN;
+	if(response == NULL) {
+		PRNT("Error: NULL Response");
+		return -1;
+	}
+	response->phrase = malloc(WEBCLIENT_CONF_MAX_PHRASE_SIZE);
+	if (response->phrase == NULL) {
+		PRNT("Error: Fail to init");
+		return -1;
+	}
+	response->message = malloc(WEBCLIENT_CONF_MAX_MESSAGE_SIZE);
+	if (response->message == NULL) {
+		PRNT("Error: Fail to init");
+		free(response->phrase);
+		return -1;
+	}
+	response->headers = malloc(sizeof(struct http_keyvalue_list_t));
+	if (response->headers == NULL || http_keyvalue_list_init(response->headers) < 0) {
+		PRNT("Error: Fail to init");
+		free(response->phrase);
+		free(response->message);
+		return -1;
+	}
+	FUNC_EX;
+	return 0;
+}
+
+void ws_test_http_client_response_release(struct http_client_response_t *response)
+{
+	FUNC_EN;
+	free(response->phrase);
+	free(response->message);
+	if (http_keyvalue_list_release(response->headers) != 0) {
+		PRNT("http_keyvalue_list_release(response->headers) failed");
+	}
+	free(response->headers);
+	FUNC_EX;
+}
+
+static int ws_test_wget_gethostip(FAR char *hostname, in_addr_t *ipv4addr)
+{
+	FUNC_EN;
+	struct hostent *he;
+	he = gethostbyname(hostname);
+	if (he == NULL) {
+		PRNT("gethostbyname failed");
+		return WGET_ERR;
+	} else if (he->h_addrtype != AF_INET) {
+		PRNT("gethostbyname returned an address of type: %d", he->h_addrtype);
+		return WGET_OK;
+	}
+
+	memcpy(ipv4addr, he->h_addr, sizeof(in_addr_t));
+	FUNC_EX;
+	return OK;
+}
+
+
+#ifdef CONFIG_NET_SECURITY_TLS
+static void ws_test_wget_tls_debug(void *ctx, int level, const char *file, int line, const char *str)
+{
+	FUNC_EN;
+	printf("%s:%04d: %s", file, line, str);
+	FUNC_EX;
+}
+
+static void ws_test_wget_tls_release(struct http_client_tls_t *client)
```

Comment:
better to move tls related functions like ws_test_wget_tls_debug, ws_test_wget_tls_release etc to a webserver_test_tls file. it will make code easy to read and maintain